tests/unit_tests/system/test_calculate_RR0.cpp

```diff
@@ -8,28 +8,23 @@
 namespace openturbine::tests {
 
 TEST(CalculateRR0Tests, OneNode) {
-    const auto r0 = Kokkos::View<double[1][1][4]>("r0");
-    constexpr auto r0_host_data = std::array{1., 2., 3., 4.};
-    const auto r0_host = Kokkos::View<const double[1][1][4], Kokkos::HostSpace>(r0_host_data.data());
-    const auto r0_mirror = Kokkos::create_mirror(r0);
-    Kokkos::deep_copy(r0_mirror, r0_host);
-    Kokkos::deep_copy(r0, r0_mirror);
-
-    const auto r = Kokkos::View<double[1][1][4]>("r");
-    constexpr auto r_host_data = std::array{5., 6., 7., 8.};
-    const auto r_host = Kokkos::View<const double[1][1][4], Kokkos::HostSpace>(r_host_data.data());
-    const auto r_mirror = Kokkos::create_mirror(r);
-    Kokkos::deep_copy(r_mirror, r_host);
-    Kokkos::deep_copy(r, r_mirror);
+    const auto x = Kokkos::View<double[1][1][7]>("x");
+    constexpr auto x_host_data = std::array{1., 2., 3., 4., 5., 6., 7.};
+    const auto x_host = Kokkos::View<const double[1][1][7], Kokkos::HostSpace>(x_host_data.data());
+    const auto x_mirror = Kokkos::create_mirror(x);
+    Kokkos::deep_copy(x_mirror, x_host);
+    Kokkos::deep_copy(x, x_mirror);
 
     const auto rr0 = Kokkos::View<double[1][1][6][6]>("rr0");
 
-    Kokkos::parallel_for("CalculateRR0", 1, CalculateRR0{0, r0, r, rr0});
+    Kokkos::parallel_for("CalculateRR0", 1, CalculateRR0{0, x, rr0});
 
-    constexpr auto expected_rr0_data = std::array{
-        2780., 4400.,  -400., 0.,     0.,    0.,    -3280., 2372., 3296., 0.,    0.,     0.,
-        2960., -1504., 4028., 0.,     0.,    0.,    0.,     0.,    0.,    2780., 4400.,  -400.,
-        0.,    0.,     0.,    -3280., 2372., 3296., 0.,     0.,    0.,    2960., -1504., 4028.};
+    constexpr auto expected_rr0_data = std::array{-44., 4.,   118., 0.,   0.,   0.,    //
+                                                  116., -22., 44.,  0.,   0.,   0.,    //
+                                                  22.,  124., 4.,   0.,   0.,   0.,    //
+                                                  0.,   0.,   0.,   -44., 4.,   118.,  //
+                                                  0.,   0.,   0.,   116., -22., 44.,   //
+                                                  0.,   0.,   0.,   22.,  124., 4.};
```

Comment:
Why did these values change?

Reply:
Instead of specifying `R` and `R0`, the new test specifies `RR0`, which is the current global rotation, as a quaternion, which is then turned into the `RR0` matrix. This was done so that the global rotation could be calculated in `calculate_QP_position` and stored for later use. The values changed because the inputs changed.